add iOS EditBox multiline support

zilongshanren opened the pull request:
When the inputMode is ANY, then we use TextView to input.
otherwise we use normal UITextField.

Fixed issue https://github.com/cocos2d/cocos2d-x/issues/13436

zilongshanren committed:
add iOS EditBox multiline support

When the inputMode is ANY, then we use TextView to input.
otherwise we use normal UITextField.

Fixed issue https://github.com/cocos2d/cocos2d-x/issues/13436

zilongshanren commented:
@Mazyod  please review this PR, thanks.

Mazyod commented:
Sure thing! Sorry, I wanted to review it this morning, but been really busy. Will make sure to review within the next 12h max.

zilongshanren commented:
@Mazyod 
Thanks. Anytime is good to me.

Mazyod committed:
[refactor] Break down the file into separate files, because it exceeded 1 kLOC :scream_cat:
[refactor] Consolidate `UITextView` and `UITextField` APIs.
[refactor] rename classes that start with `UI` to `CCUI`, since that prefix is reserved by Apple.
[bugfix] setting the placeholder for `CCUISingleLineTextField` was missing in `createSingleLineTextField`
[bugfix] In `MultiLineTextView`, if `placeholder` changed, there doesn't seem to be a deterministic way of updating the `placeholderLabel`. We are relying on `drawRect:`. Instead, `placeholder` is now a computed property on top of `placeholderLabel.text`.
[bugfix] If the user wants multi-line secure input, we should set it, since it is possible, right?
[minor] `[viewWithTag:999]` was used instead of calling the property itself.
[minor] `drawRect:` was used to load properties, when it is normally in the getter.
[minor] fix ObjC code style

Mazyod committed:
Oops, committed duplicate code by mistake

zilongshanren committed:
Merge pull request #27 from Mazyod/addiOSMultiline-support

UIEditBox iOS changes

zilongshanren committed:
Merge branch 'v3' into addiOSMultiline-support

# Conflicts:
#	build/cocos2d_libs.xcodeproj/project.pbxproj
